otx/cli/utils/tests.py

```diff
@@ -763,3 +764,36 @@ def otx_train_auto_config(root, otx_dir: str, args: Dict[str, str]):
     command_line.extend(["--work-dir", f"{work_dir}"])
     command_line.extend(args["train_params"])
     check_run(command_line)
+
+
+def otx_regression_testing(template, root, otx_dir, args, criteria, threshold=0.05):
+    template_work_dir = get_template_dir(template, root)
+
+    command_line = [
+        "otx",
+        "eval",
+        template.model_template_path,
+        "--test-data-roots",
+        f'{os.path.join(otx_dir, args["--test-data-roots"])}',
+        "--load-weights",
+        f"{template_work_dir}/trained_{template.model_template_id}/weights.pth",
+        "--save-performance",
+        f"{template_work_dir}/trained_{template.model_template_id}/performance.json",
+    ]
+    command_line.extend(["--work-dir", f"{template_work_dir}"])
+    command_line.extend(args.get("eval_params", []))
+    check_run(command_line)
+
+    performance_json_path = f"{template_work_dir}/trained_{template.model_template_id}/performance.json"
+    assert os.path.exists(performance_json_path)
+
+    with open(performance_json_path) as read_file:
+        trained_performance = json.load(read_file)
+
+    model_criteria = criteria[template.name]
+    modified_criteria = model_criteria - (model_criteria * threshold)
+    for k in trained_performance.keys():
+        print(f"trained_performance: {trained_performance[k]}, modified_criteria:{modified_criteria}")
+        assert (
+            trained_performance[k] <= modified_criteria
```

Comment:
As I see that criteria checks that model performance <= 0.95?
Why? I thought we will train all  our templates and record exact numbers (or small performance range).
why do we need: `model_criteria - (model_criteria * threshold)` as we always store model_criteria as 1.0 in `json`.
I don't understand what this test checks

Reply:
@kprokofi , This PR is still WIP. 
For the current status, since we didn't know the current model's performance, we need to check it first.
So, I want to make FAIL for all test cases. That's the reason why I set the 1.0 value. 

After measuring the current model's performance, I'll edit the criteria value. Thanks for question !